src/main/java/mate/controller/AddCarController.java

```diff
@@ -0,0 +1,37 @@
+package mate.controller;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.service.CarService;
+import mate.service.ManufacturerService;
+
+public class AddCarController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate");
+    private final CarService carService = (CarService) injector.getInstance(CarService.class);
+    private final ManufacturerService manufacturerService = (ManufacturerService)
+            injector.getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/cars/add.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Car car = new Car();
+        car.setModel(req.getParameter("model"));
+        Long manufacturerId = Long.parseLong((req.getParameter("manufacturer_id")));
+        car.setManufacturer(manufacturerService.get(manufacturerId));

```

Comment:
what if you don't have manufacturer with this id? you can add one more method in  ManufacurerService which returns Optional of manufacturer and in case it's empty- add message to jsp page that manufacurer does not exist

Reply:
Remember that you can pass id of a manufacturer, car, or driver through the form where your models depend on one another. You can expect that the entity with this id already exists in the DB.